app/service/file_svc.py

```diff
@@ -70,6 +72,15 @@ async def create_exfil_sub_directory(self, dir_name):
             os.makedirs(path)
         return path
 
+    async def create_exfil_operation_directory(self, dir_name,agent_name):
+        op_list = self.data_svc.ram['operations']
+        op_list_filtered = [x for x in op_list if x.state not in x.get_finished_states()]
+        agent_opid = [(x.name,' ',x.start.strftime("%Y-%m-%dT%H:%M:%SZ").replace("T"," @ ")) for x in op_list_filtered if agent_name in [y.paw for y in x.agents]]
```

Comment:
This will cause issues on filesystems that consider ':' an illegal character. Also I'd shy away from using spaces in paths unless it was absolutely necessary.

Reply:
Also linter errors are present here

Reply:
AFAIK, caldera does not limit the operation name. Since this code pulls from operation name given by the end user, and the user could possibly name it something with illegal characters, I will parse for illegal characters and spaces and replace them to underscore.

Reply:
User input was also definite a concern but I was just commenting on the colons within the timestamp.

Reply:
Tested using an operation with name `A:<>"/\|?*B`. Characters and white space corrects to underscore. Consolidated timestamp.

<img width="328" alt="image" src="https://user-images.githubusercontent.com/81784737/162098088-9f74011f-4adb-4df7-95fd-fabfa2d357b3.png">